client/starwhale/core/dataset/store.py

```diff
@@ -408,6 +415,30 @@ def _make_file(self, bucket: str, key_compose: str) -> FileLikeObj:
             return io.BytesIO(f.read(_end - _start + 1))
 
 
+class SignedUrlBackend(StorageBackend, CloudRequestMixed):
+    def __init__(self, dataset_uri: URI) -> None:
+        super().__init__(kind=SWDSBackendType.SignedUrl)
+        self.dataset_uri = dataset_uri
+
+    def _make_file(self, auth: str, key_compose: t.Tuple[str, int, int]) -> FileLikeObj:
+        _key, _start, _end = key_compose
+        url = self.sign_uri(_key, auth)
+        headers = {}
+        if _start and _end:
+            headers = {"Range": f"bytes={_start}-{_end}"}
+        r = requests.get(url, headers=headers)
+        return io.BytesIO(r.content)
+
+    def sign_uri(self, uri: str, auth_name: str) -> t.Any:
+        r = self.do_http_request(
```

Comment:
add timeout? requests default timeout is None, will blocking forever.

Reply:
`self.do_http_request` has default timeout scends